src/RequestParser.jl

```diff
@@ -48,24 +48,37 @@ end
 # this: https://github.com/joyent/node/blob/master/src/node_http_parser.cc#L207
 
 function on_header_field(parser, at, len)
-    r = pd(parser).request
+    par = pd(parser)
+    if par.num_fields == par.num_values
+        par.num_fields += 1
+        push!(par.vec_fields, "")
+    end
     header = unsafe_string(convert(Ptr{UInt8}, at))
     header_field = header[1:len]
-    r.headers["current_header"] = header_field
+    par.vec_fields[par.num_fields] = string(par.vec_fields[par.num_fields], header_field)
     return 0
 end
 
 function on_header_value(parser, at, len)
-    r = pd(parser).request
+    par = pd(parser)
+    if par.num_values != par.num_fields
+        par.num_values += 1
+        push!(par.vec_values, "")
+    end
     s = unsafe_string(convert(Ptr{UInt8}, at), Int(len))
-    r.headers[r.headers["current_header"]] = s
-    r.headers["current_header"] = ""
-    # delete!(r.headers, "current_header")
+    header_value = s[1:len]
+    par.vec_values[par.num_values] = string(par.vec_values[par.num_values], header_value)
     return 0
 end
 
 function on_headers_complete(parser)
-    r = pd(parser).request
+    par = pd(parser)
+    r = par.request
+    merge!(r.headers, Dict(zip(par.vec_fields, par.vec_values)))
```

Comment:
Note that this isn't strictly correct. By calling `zip` on your `par.vec_fields`, you're inadvertently dropping any duplicate header fields, such as `Set-Cookie`, which is commonly sent as several separate header fields. Not that this was being handled correctly before or necessarily needs to be addressed here, but it was something I remember fixing in the port of everything to HTTP.jl.